spring-content-s3/src/main/java/internal/org/springframework/content/s3/store/DefaultS3StoreImpl.java

```diff
@@ -116,10 +117,15 @@ public Resource getResource(S entity, PropertyPath propertyPath) {
             return null;
 
         S3ObjectId s3ObjectId = null;
-        if (placementService.canConvert(property.getContentIdType(entity).getClass(), S3ObjectId.class)) {
+        if (placementService.canConvert(property.getContentIdType(entity).getObjectType(), S3ObjectId.class)) {
             s3ObjectId = placementService.convert(property.getContentId(entity), S3ObjectId.class);
```

Comment:
### First Question:
Are Entities allowed to have contentIds of type S3ObjectId or of other custom types?

The latter case appears to be covered: for custom id type, that has `<custom id class> -> S3ObjectId`-converter registered the bucket name will be overridden by `@Bucket` value. Otherwise there should be an error later in getResouce(id).

But if contentId is already of type `S3ObjectId` there is another bug:
It will be actually first converted by `Converter<Serializable, S3ObjectId>`, which results in object storage key being equal to S3ObjectId::toString(). And after that bucket will be overridden by value in `@Bucket` field (This is correct, but may be confusing for users who expect that bucket name set in contentId has precedence over `@Bucket` field. But this is a rare case and it does not really make sense for an entity to have both S3ObjectId contentId and `@Bucket` field at the same time).

### Second Question:
I noticed that s3 implementation has two default converters `Converter<Serializable, S3ObjectId>` and `Converter<Object, S3ObjectId>`, while GCP and Azure have only one `Serializable->BlobId`-converter each. @paulcwarren , can you please tell me why this is the case? Why for each backend one default `Object(since Entities are not necessarily Serializable)->S3ObjectId/BlobId`-converter that can convert both ids and entities is not enough? I think there might be bugs here too (for example, for GCP and Azure the default converter is only called on the whole entity (and `@Bucket` value can be taken into account), if entity is Serializable, even in `getResource(S entity)`.